shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -0,0 +1,588 @@
+        )
+        if location not in (locations | child_locations):
+            message = self.msg_store.location_content_unable_to_transfer(location)
+            return self._response_for_select_location(message=message)
+
+    def _scan_location__check_stock(self, location):
+        quants_in_location = self.env["stock.quant"].search(
+            [("location_id", "=", location.id)]
+        )
+        if not quants_in_location:
+            message = self.msg_store.location_empty(location)
+            return self._response_for_select_location(message=message)
+
+    def _scan_product__scan_product(self, location, barcode):
+        search = self._actions_for("search")
+        product = search.product_from_scan(barcode)
+        handlers = [
+            self._scan_product__check_tracking,
+            self._scan_product__select_move_line,
+            # If no line is found, we might try to create one,
+            # if allow_move_create is True
+            self._scan_product__check_create_move_line,
+            # And unreserve other moves before, if option allow_unreserve_other_moves
+            # is set to True.
+            self._scan_product__unreserve_move_line,
+            self._scan_product__create_move_line,
+        ]
+        if product:
+            response = self._use_handlers(handlers, product, location)
+            if response:
+                return response
+
+    def _scan_product__check_tracking(self, product, location, lot=None):
+        if product.tracking == "lot":
+            message = self.msg_store.scan_lot_on_product_tracked_by_lot()
+            return self._response_for_select_product(location, message=message)
+
+    def _scan_product__select_move_line_domain(self, product, location, lot=None):
+        domain = [
+            ("location_id", "=", location.id),
+            ("product_id", "=", product.id),
+            ("state", "in", ("assigned", "partially_available")),
+            ("shopfloor_user_id", "=", False),
+            ("picking_id.picking_type_id", "in", self.picking_types.ids),
+        ]
+        if lot:
+            lot_domain = [("lot_id", "=", lot.id)]
+            domain = AND([domain, lot_domain])
+        return domain
+
+    def _scan_product__select_move_line(self, product, location, lot=None):
+        domain = self._scan_product__select_move_line_domain(product, location, lot=lot)
+        move_line = self.env["stock.move.line"].search(domain, limit=1)
+        if move_line:
+            stock = self._actions_for("stock")
+            if self.work.menu.no_prefill_qty:
+                # First, mark move line as picked with qty_done = 0,
+                # so the move wont be split because 0 < qty_done < product_uom_qty
+                stock.mark_move_line_as_picked(move_line, quantity=0)
+                # Then, set the no prefill qty on the move line
+                if lot:
+                    qty_done = 1
+                else:
+                    qty_done = 1
```

Comment:
missing product packaging case where qty_done = packaging.qty

Reply:
What I did/changed regarding packaging scanning:

- without `no_prefill_qty` option, when we scan an item (product/lot/packaging) on the `scan_product` state, the `qty_done` is always set to the `product_uom_qty`. If we have the `allow_move_create` option enabled and that we scan a packaging, the move created will have a qty == packaging.qty even if we have more available in the location.
- with `no_prefill_qty` option, when we scan a product/lot/packaging, the initual `qty_done` will be 0 (it was 1 before), so we wait the `set_quantity` state to increase the qty on the move line, but that means we have to scan the product/lot/packaging twice... which is maybe not what we want :thinking: 

I can revert the change on the initial value of `qty_done` with `no_prefill_qty` enabled, you tell me.

Reply:
I updated the PR to restore the initial behavior. Should be fixed.